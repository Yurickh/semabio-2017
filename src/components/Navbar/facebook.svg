<?xml version="1.0" encoding="UTF-8"?>
<svg width="7px" height="15px" viewBox="0 0 7 15" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 45.2 (43514) - http://www.bohemiancoding.com/sketch -->
    <title>Page 1</title>
    <desc>Created with Sketch.</desc>
    <defs>
        <polygon id="path-1" points="3.49803548 14.8583284 0 14.8583284 0 4.47761194e-05 3.49803548 4.47761194e-05 6.99607097 4.47761194e-05 6.99607097 14.8583284"></polygon>
    </defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="BetterDesk" transform="translate(-1099.000000, -84.000000)">
            <g id="Page-1" transform="translate(1099.000000, 84.000000)">
                <mask id="mask-2" fill="white">
                    <use xlink:href="#path-1"></use>
                </mask>
                <g id="Clip-2"></g>
                <path d="M4.60861935,4.46198507 L4.60861935,3.27519403 C4.60861935,2.69108955 4.99903871,2.5578806 5.27429677,2.5578806 L6.95768387,2.5578806 L6.95768387,0.0128059701 L4.64068387,4.47761194e-05 C2.07371613,4.47761194e-05 1.48503871,1.90414925 1.48503871,3.11646269 L1.48503871,4.46198507 L-9.03225806e-05,4.46198507 L-9.03225806e-05,6.25795522 L-9.03225806e-05,7.43243284 L1.49790968,7.43243284 L1.49790968,14.8583284 L4.49345806,14.8583284 L4.49345806,7.43243284 L6.7208129,7.43243284 L6.82965161,6.26444776 L6.99607097,4.46198507 L4.60861935,4.46198507 Z" id="Fill-1" fill="#FEFEFE" mask="url(#mask-2)"></path>
            </g>
        </g>
    </g>
</svg>