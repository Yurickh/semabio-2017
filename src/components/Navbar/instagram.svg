<?xml version="1.0" encoding="UTF-8"?>
<svg width="15px" height="15px" viewBox="0 0 15 15" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 45.2 (43514) - http://www.bohemiancoding.com/sketch -->
    <title>Page 1</title>
    <desc>Created with Sketch.</desc>
    <defs>
        <polygon id="path-1" points="14.9274468 14.8943556 14.9274468 0.0720422535 0 0.0720422535 0 14.8943556 14.9274468 14.8943556"></polygon>
    </defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="BetterDesk" transform="translate(-1122.000000, -84.000000)">
            <g id="Page-1" transform="translate(1122.000000, 84.000000)">
                <g id="Group-3" transform="translate(0.000000, 0.033592)">
                    <mask id="mask-2" fill="white">
                        <use xlink:href="#path-1"></use>
                    </mask>
                    <g id="Clip-2"></g>
                    <path d="M4.24856383,1.34248944 C2.61132979,1.34248944 1.27941489,2.66502465 1.27941489,4.29051761 L1.27941489,10.6756585 C1.27941489,12.3013627 2.61132979,13.6240035 4.24856383,13.6240035 L10.6787766,13.6240035 C12.3160106,13.6240035 13.6479255,12.3013627 13.6479255,10.6756585 L13.6479255,4.29051761 C13.6479255,2.66502465 12.3160106,1.34248944 10.6787766,1.34248944 L4.24856383,1.34248944 Z M10.6787766,14.8943556 L4.24856383,14.8943556 C1.90590426,14.8943556 -5.31914894e-05,13.0019261 -5.31914894e-05,10.6756585 L-5.31914894e-05,4.29051761 C-5.31914894e-05,1.96446127 1.90590426,0.0720316901 4.24856383,0.0720316901 L10.6787766,0.0720316901 C13.0214362,0.0720316901 14.9275,1.96446127 14.9275,4.29051761 L14.9275,10.6756585 C14.9275,13.0019261 13.0214362,14.8943556 10.6787766,14.8943556 L10.6787766,14.8943556 Z" id="Fill-1" fill="#FFFFFF" mask="url(#mask-2)"></path>
                </g>
                <path d="M7.46373404,4.99203169 C6.06160638,4.99203169 4.92107447,6.12463732 4.92107447,7.51678521 C4.92107447,8.9089331 6.06160638,10.0414331 7.46373404,10.0414331 C8.86575532,10.0414331 10.0062872,8.9089331 10.0062872,7.51678521 C10.0062872,6.12463732 8.86575532,4.99203169 7.46373404,4.99203169 M7.46373404,11.3119965 C5.35607447,11.3119965 3.6415,9.60949648 3.6415,7.51678521 C3.6415,5.42407394 5.35607447,3.72146831 7.46373404,3.72146831 C9.57118085,3.72146831 11.2858617,5.42407394 11.2858617,7.51678521 C11.2858617,9.60949648 9.57118085,11.3119965 7.46373404,11.3119965" id="Fill-4" fill="#FFFFFF"></path>
                <path d="M12.3683723,3.50444366 C12.3683723,4.0083169 11.9570957,4.41648592 11.4497553,4.41648592 C10.9425213,4.41648592 10.5310319,4.0083169 10.5310319,3.50444366 C10.5310319,3.00067606 10.9425213,2.59208451 11.4497553,2.59208451 C11.9570957,2.59208451 12.3683723,3.00067606 12.3683723,3.50444366" id="Fill-6" fill="#FFFFFF"></path>
            </g>
        </g>
    </g>
</svg>