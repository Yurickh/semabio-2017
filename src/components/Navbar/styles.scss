@import 'ambience/typography.scss';
@import 'ambience/responsive.scss';

.navbar {
  position: relative;
  z-index: 2;

  display: flex;
  align-items: center;
  justify-content: space-between;

  width: 70vw;
  margin: 0 auto;
  padding-top: 32px;

  @include small-only {
    width: 82vw;
  }

  & > .brand {
    display: inline-flex;
    align-items: center;

    cursor: pointer;
  }

  & > .brand > .name {
    font-weight: $weight-bold;

    margin-left: 12px;

    text-transform: uppercase;

    color: #FFF;
  }

  &.-open > .brand {
    display: none;
  }

  & > .links {
    position: absolute;
    top: 100px;
    right: 0;

    display: flex;
    flex-direction: column;
    align-items: flex-end;
  }

  & > .links > .link {
    font-size: 24px;
    line-height: 56px;
    text-transform: uppercase;
  }
}
