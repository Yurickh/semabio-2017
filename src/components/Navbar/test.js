import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import Navbar from '../Navbar'

describe('Navbar', () => {
  it('should render correctly', () => {
    const navbar = shallow(<Navbar />)
    expect(toJSON(navbar)).toMatchSnapshot()
  })
})
