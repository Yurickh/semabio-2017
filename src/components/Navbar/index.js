import React, { Component } from 'react'
import PropTypes from 'prop-types'

import NavBurger from '../NavBurger'
import { Link } from 'react-scroll'

import semabioLogo from './semabio-logo.svg'

import './styles.css'

class Navbar extends Component {
  variant = () => {
    return this.props.open ? '-open' : ''
  }

  render() {
    return (
      <nav className={ `navbar ${this.variant()}` }>
        <a href="/" className="brand">
          <img
            alt="logo"
            className="logo"
            src={semabioLogo}
          />
          <span className="name">
            Semana da Biologia
          </span>
        </a>
        <div className="social">
          <a className="icon -facebook"
            target="_blank"
            rel="noopener noreferrer"
            href="https://www.facebook.com/SemaBio/"
          >
          </a>
          <a className="icon -instagram"
            target="_blank"
            rel="noopener noreferrer"
            href="https://www.instagram.com/semabiounb/"
          >
          </a>
          <NavBurger />
        </div>
      </nav>
    )
  }
}

Navbar.propTypes = {
  open: PropTypes.bool,
  toggleMenu: PropTypes.func.isRequired,
}

Navbar.defaultProps = {
  open: false
}

export default Navbar
