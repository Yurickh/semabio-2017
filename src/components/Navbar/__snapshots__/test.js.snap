// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
<nav
  className="navbar"
>
  <div
    className="brand"
  >
    <img
      alt="logo"
      className="logo"
      src="semabio-logo.svg"
    />
    <span
      className="name"
    >
      Semana da Biologia
    </span>
  </div>
</nav>
`;
